### ISSUE TITLE
combo for DALI's GDS with NVIDIA/apex

### ISSUE THREAD

**Issue by 4fee8fea** [NVIDIA/DALI#3747]
As [Numpy Reader - GPUDirect Storage Support](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/general/data_loading/numpy_reader.html) mentions,

> In order to enable GDS support in DALI, make sure GDS is installed. On Systems with CUDA 11.4 or newer, GDS is already installed as part of the CUDA 11.4 SDK. For older CUDA releases, please install GDS separately (follow the link above for instructions).

While using `DALI`'s `GDS`, I also wanted to use the `NVIDIA/apex` technology, but its installation process reported the following error,

```
RuntimeError: Cuda extensions are being compiled with a version of Cuda that does not match the version used to compile Pytorch binaries.  Pytorch binaries were compiled with Cuda 11.3.
    In some cases, a minor-version mismatch will not cause later errors:  https://github.com/NVIDIA/apex/pull/323#discussion_r287021798.  You can try commenting out this check (at your own risk).
    Running setup.py install for apex ... error
```

However, the highest Cuda version currently supported by PyTorch is 11.3. 

It seems that the Cuda versions required by `GDS` and `apex` are diverging. Is there a way to use both GDS and apex at the current time point?

**Comment by JanuszL** [comment]
Hi @4fee8fea,

I think it would be best to ask on [the related dev forum](https://forums.developer.nvidia.com/c/accelerated-computing/cuda).
What I can personally suggest is to install two CUDA toolkits, use 11.3 to build APEX and 11.4 to install GDS but it may require a bit of experimentation.

**Comment by 4fee8fea** [comment]
For Cuda 11.6, the GDS can now be used directly. I will use Cuda 11.3 to build the apex and switch to Cuda 11.6 for GDS. Thanks for your suggestions!!

**Comment by 4fee8fea** [comment]
Hi @JanuszL,

In order to be able to use `apex` and `GDS` at the same time, I installed both cuda11.3 and cuda11.6 on my server and switched between them via a soft link. 

After checking that the soft link and results of command `nvcc` are correct, I set the `device` parameter of NumpyReader to `gpu`. But it runs with the following error,

```
RuntimeError: Critical error when building pipeline:
Error when constructing operator: readers__Numpy, instance name: "__Numpy_0", encountered:
cuFile error (5011): cufile: CUDA Driver API error
Current pipeline object is no longer valid.
```

Previously, when only cuda11.6 was available, the setting to `gpu` worked. What could be the cause of this? Thank you very much!
